### PR TITLE
build: upgrade to typescript 3.2.2

### PR DESCRIPTION
**Pull request by JiaLiPassion**

### CODE REVIEW

File: lib/rxjs/rxjs.ts [hunk 1]
```diff
@@ -145,26 +145,28 @@ import {Observable, Subscriber, Subscription} from 'rxjs';
     Subscriber.prototype.next = function() {
       const currentZone = Zone.current;
       const subscriptionZone = this._zone;
+      const args: [any?] = Array.prototype.slice.call(arguments) as [any?];
```
Comment (mhevery) [inline lib/rxjs/rxjs.ts]: This seems like extra work which we have to do. Why is the `slice` needed? Can't we just cast it somehow to make TS happy?

File: lib/rxjs/rxjs.ts [hunk 2]
```diff
@@ -145,26 +145,28 @@ import {Observable, Subscriber, Subscription} from 'rxjs';
...
       } else {
-        return next.apply(this, arguments);
+        return next.apply(this, args);
       }
     };
 
     Subscriber.prototype.error = function() {
       const currentZone = Zone.current;
       const subscriptionZone = this._zone;
+      const args: [any?] = Array.prototype.slice.call(arguments) as [any?];
```
Comment (mhevery) [inline lib/rxjs/rxjs.ts]: same...

File: lib/rxjs/rxjs.ts (hunk 1 above)
Comment (JiaLiPassion) [reply lib/rxjs/rxjs.ts]: Sure,  without making a new `slice`, just `cast arguments to any`.

File: lib/rxjs/rxjs.ts (hunk 2 above)
Comment (JiaLiPassion) [reply lib/rxjs/rxjs.ts]: Got it.

**Pull request merged**